### PR TITLE
reorganize tests, make devdeps testing optional, and unpin latest Python testing

### PR DESCRIPTION
**Pull request by zacharyburnett**
Resolves [SCSB-79](https://jira.stsci.edu/browse/SCSB-79)

This PR reorganizes CI tests and moves devdeps tests to a new optional workflow that can be started with the label `run devdeps tests`

**Checklist**
- [x] ~added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)~
- [x] updated relevant tests
- [x] ~updated relevant documentation~
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: .github/workflows/ci_cron.yml [hunk 1]
```diff
@@ -1,18 +1,25 @@
-name: Weekly cron
+name: scheduled tests
```
Comment (zacharyburnett) [inline .github/workflows/ci_cron.yml:1]:
```suggestion
name: test on schedule
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,18 +1,17 @@
-name: CI
+name: test
 
 on:
   push:
     branches:
-      - main
-      - '*x'
+      - master
```
Comment (kmacdonald-stsci) [inline .github/workflows/ci.yml:6]: Is this correct?  STCAL has a `main` branch, not a `master` branch.

File: .github/workflows/tests_devdeps.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: test with development versions
+
+on:
+  push:
+    branches:
+      - master
```
Comment (kmacdonald-stsci) [inline .github/workflows/tests_devdeps.yml:6]: Is this correct?  STCAL has a `main` branch, not a `master` branch.

**Review by kmacdonald-stsci — COMMENTED** [review]
Some of the files listed a `master` branch, but STCAL uses a `main` branch.  Is the use of `master` in this PR correct?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
     tags:
       - '*'
   pull_request:
-    branches:
-      - main
   schedule:
     # Weekly Monday 9AM build
-    - cron: "0 9 * * 1"
+    # * is a special character in YAML so you have to quote this string
+    - cron: '0 9 * * 1'
```
Comment (hbushouse) [inline .github/workflows/ci.yml:14]: Is it OK to have jobs for multiple repos all running at the same time, e.g. Monday 6am and 9am?

File: .github/workflows/ci_cron.yml [hunk 5]
```diff
@@ -1,18 +1,25 @@
-name: Weekly cron
+name: test on schedule
 
 on:
   schedule:
     # Weekly Monday 6AM build
     - cron: "0 0 * * 1"
```
Comment (hbushouse) [inline .github/workflows/ci_cron.yml:6]: "0 0" is midnight, right? But comment says 6AM.

File: .github/workflows/tests_devdeps.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      - run: tar -xzvf tmp/minimal-webbpsf-data.tar.gz -C tmp/data/
+      - id: data_hash
+        run: echo "hash=${{ hashFiles( 'tmp/data' ) }}" >> $GITHUB_OUTPUT
+      - run: mv tmp/data/* ${{ steps.data.outputs.path }}
+      - uses: actions/cache@v3
+        with:
+          path: ${{ steps.data.outputs.path }}
+          key: data-${{ steps.data_hash.outputs.hash }}
+      - id: webbpsf_path
+        run: echo "path=${{ steps.data.outputs.path }}/webbpsf-data" >> $GITHUB_OUTPUT
```
Comment (zacharyburnett) [inline .github/workflows/tests_devdeps.yml:57]:
```suggestion
```

File: .github/workflows/tests_devdeps.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      setenv: |
+        WEBBPSF_PATH: ${{ needs.data.outputs.webbpsf_path }}
+        CRDS_PATH: /tmp/crds_cache
+        CRDS_CLIENT_RETRY_COUNT: 3
+        CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
+      cache-path: ${{ needs.data.outputs.data_path }}
+      cache-key: data-${{ needs.data.outputs.data_hash }}
+      envs: |
+        - linux: py3-jwst-devdeps-xdist
+        - linux: py3-romancal-devdeps-xdist
```
Comment (zacharyburnett) [inline .github/workflows/tests_devdeps.yml:72]:
```suggestion
    with:
      setenv: |
        CRDS_PATH: /tmp/data/crds_cache
        CRDS_CLIENT_RETRY_COUNT: 3
        CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
      envs: |
        - linux: py3-jwst-devdeps-xdist
        - linux: py3-romancal-devdeps-xdist
```

File: .github/workflows/ci_cron.yml [hunk 8]
```diff
@@ -1,16 +1,24 @@
...
     uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
     with:
       envs: |
         - macos: py39-xdist
+        - windows: py39-xdist
         - macos: py310-xdist
-        - linux: py3-devdeps-xdist
+        - windows: py310-xdist
+        - windows: py311-xdist
+        - windows: py3-xdist
```
Comment (zacharyburnett) [inline .github/workflows/ci_cron.yml:24]:
```suggestion
```

**Comment by hbushouse** [comment]
github doesn't want to let me create a review, despite the fact that I've been tagged as a reviewer.

**Review by hbushouse — APPROVED** [review]
LGTM

**Pull request merged**